File: Introduction_to_numerical_programming_using_Python_and_CPP_Beu/Ch02/Python/P02-Fact_recur.py
# Factorial based on recursivity

#============================================================================
def FactRec(n):
#----------------------------------------------------------------------------
#  Returns the factorial of n using recursivity
#----------------------------------------------------------------------------
   return (n * FactRec(n-1) if (n > 1) else 1e0)

# main

n = int(input("n = "))

print("{0:d}! = {1:f}".format(n,FactRec(n)))

File: Introduction_to_numerical_programming_using_Python_and_CPP_Beu/Ch02/C/P02-Fact_recur.cpp
// Factorial based on recursivity
#include <stdio.h>

//===========================================================================
double FactRec(int n)
//---------------------------------------------------------------------------
// Returns the factorial of n using recursivity
//---------------------------------------------------------------------------
{
   return ((n > 1) ? n * FactRec(n-1) : 1e0);
}

int main()
{
   int n;

   printf("n = "); scanf("%d",&n);

   printf("%d! = %f\n",n,FactRec(n));
}
